addons/binding/org.openhab.binding.samsungtv/src/main/java/org/openhab/binding/samsungtv/internal/handler/SamsungTvHandler.java

```diff
@@ -320,20 +257,24 @@ private synchronized void createService(RemoteDevice device) {
                     if (newService != null) {
                         if (existingService != null) {
                             stopService(existingService);
+                            startService(newService);
+                            logger.debug("Restarting service in UPnP mode for: {}, {} ({})", modelName, type, udn);
+                        } else {
+                            startService(newService);
+                            logger.debug("Started service for: {}, {} ({})", modelName, type, udn);
                         }
-
-                        startService(newService);
+                    } else {
+                        logger.trace("Skipping unknown UPnP service: {}, {} ({})", modelName, type, udn);
                     }
+
                 } else {
-                    logger.debug("Device rediscovered, clear caches");
+                    logger.debug("Service rediscoved, clearing caches: {}, {} ({})", modelName, type, udn);
                     existingService.clearCache();
                 }
                 putOnline();
-            } else {
-                logger.debug("Ignore device={}", device);
             }
         } else {
-            logger.debug("Thing not yet initialized");
+            logger.error("Thing not yet initialized");
```

Comment:
Is there something tremendously wrong in the setup of a user that this should be logged on error now? 
See [logging guidelines](https://www.openhab.org/docs/developer/development/guidelines.html#e-logging).